benchmarks/gems_master_model/benchmark_amoebanet_gems_master.py

```diff
@@ -0,0 +1,292 @@
+            )
+            amoebanet_shapes_list.append(x)
+
+model_gen.shape_list = amoebanet_shapes_list
+
+logging.info(f"Shape of model on local_rank {local_rank} : {model_gen.shape_list}")
+
+del model_gen
+del model
+torch.cuda.ipc_collect()
+
+model = amoebanet.amoebanetd(
+    num_classes=num_classes, num_layers=args.num_layers, num_filters=args.num_filters
+)
+
+# GEMS Model 1
+model_gen1 = model_gen = model_generator(
+    model=model,
+    split_size=mp_size,
+    input_size=(int(batch_size / parts), 3, image_size, image_size),
+    balance=balance,
+    shape_list=amoebanet_shapes_list,
+)
+model_gen1.ready_model(split_rank=local_rank)
+
+model = amoebanet.amoebanetd(
+    num_classes=num_classes, num_layers=args.num_layers, num_filters=args.num_filters
+)
+
+# GEMS Model 2
+model_gen2 = model_gen = model_generator(
+    model=model,
+    split_size=mp_size,
+    input_size=(int(batch_size / parts), 3, image_size, image_size),
+    balance=balance,
+    shape_list=amoebanet_shapes_list,
+)
+model_gen2.ready_model(split_rank=mp_size - local_rank - 1)
+
+tm_master = train_model_master(
+    model_gen1,
+    model_gen2,
+    local_rank,
+    batch_size,
+    epoch,
+    criterion=None,
+    optimizer=None,
+    parts=parts,
+    ASYNC=ENABLE_ASYNC,
+)
+
+sync_allreduce = gems_comm.SyncAllreduce(mpi_comm)
+
+############################## Dataset Definition ##############################
+
+transform = transforms.Compose(
+    [transforms.ToTensor(), transforms.Normalize((0.5, 0.5, 0.5), (0.5, 0.5, 0.5))]
+)
+torch.manual_seed(0)
+
+if APP == 1:
+    trainset = torchvision.datasets.ImageFolder(
+        datapath,
+        transform=transform,
+        target_transform=None,
+    )
+    my_dataloader = torch.utils.data.DataLoader(
+        trainset,
+        batch_size=times * batch_size,
+        shuffle=True,
+        num_workers=0,
+        pin_memory=True,
+    )
+    size_dataset = len(my_dataloader.dataset)
+elif APP == 2:
+    transform = transforms.Compose(
+        [
+            transforms.Resize((512, 512)),
+            transforms.ToTensor(),
+            transforms.Normalize((0.5, 0.5, 0.5), (0.5, 0.5, 0.5)),
+        ]
+    )
+    torch.manual_seed(0)
+    trainset = torchvision.datasets.CIFAR10(
+        root=datapath, train=True, download=True, transform=transform
+    )
+    my_dataloader = torch.utils.data.DataLoader(
+        trainset,
+        batch_size=times * batch_size,
+        shuffle=False,
+        num_workers=0,
+        pin_memory=True,
+    )
+    size_dataset = 50000
+else:
+    my_dataset = torchvision.datasets.FakeData(
+        size=10 * batch_size,
+        image_size=(3, image_size, image_size),
+        num_classes=num_classes,
+        transform=transform,
+        target_transform=None,
+        random_offset=0,
+    )
+    my_dataloader = torch.utils.data.DataLoader(
+        my_dataset,
+        batch_size=batch_size * times,
+        shuffle=False,
+        num_workers=0,
```

Comment:
expose `num_workers` as an arg unless we have a good reason to force it here

Reply:
Are you suggesting removing num_workers from the argument list or using it as a command-line argument?

Reply:
Don't remove `num_workers` as an arg. It should still be a function arg and default to zero, but I want the user to be able to pass it in the benchmark `args` similar to `args.split_size` https://github.com/OSU-Nowlab/MPI4DL/pull/11/files#diff-68cd23c73cb37b7f0b9ae6cdde9b4d13819b4461ac67588399977b94cc1f528bR65

Reply:
Done.